spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageProtocolResolver.java

```diff
@@ -84,7 +89,14 @@ public Resource resolve(String location, ResourceLoader resourceLoader) {
 							+ "so defaults are used.");
 				}
 			}
-			return new GoogleStorageResource(this.storage, location, this.settings.isAutoCreateFiles());
+
+			if (pathParts.length < 2) {
```

Comment:
gs://bucket/ would work fine, gs:// would try to create a bucket with "" as the name, which would get you an error from the GCS functions. gs:/// would cause an array out of bounds exception when i try to get that bucket name after splitting the string. I'll add a check for that last case.